.github/workflows/ci.yaml

```diff
@@ -21,16 +21,21 @@ jobs:
 
     steps:
       - name: Checkout code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4.2.2
 
       - name: Set up Python
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v5.6
         with:
           python-version: ${{matrix.python-version}}
 
       - name: Install dependencies
-        run: python -m pip install pytest pytest-asyncio langchain-openai .
+        run: |
```

Comment:
pinning to the versions in the `poetry.lock` in CI seemed advisable, which motivated this change